R/salmon.R

```diff
@@ -24,10 +24,18 @@ salmon_counts <- function(x, tx2gene = NULL) {
     all(colnames(tx2gene) == c("tx_name", "gene_id")),
     msg = "The tx2gene df object has incorrect column names."
   )
-
-  txi_salmon <- tximport::tximport(files = x, type = "salmon", tx2gene = tx2gene)
-  counts <- txi_salmon[["counts"]] |>
-    tibble::as_tibble(rownames = "rowname", .name_repair = make.names) |>
-    dplyr::rename(count = .data$X)
+  # check if gene level counts are provided or transcript level
+  if (grepl("genes.sf", basename(x), fixed = TRUE)) {
+    counts <- read.table(x, sep="\t", as.is=TRUE, header=TRUE) |>
+      dplyr::select(Name, NumReads) |>
+      dplyr::rename(rowname = Name, count = NumReads) |>
+      dplyr::mutate(rowname = gsub("\\..*", "", rowname))
+  } else {
+    txi_salmon <- tximport::tximport(files = x, type = "salmon", tx2gene = tx2gene)
+    counts <- txi_salmon[["counts"]] |>
+      tibble::as_tibble(rownames = "rowname", .name_repair = make.names) |>
+      dplyr::rename(count = .data$X) |>
+      dplyr::mutate(rowname = gsub("\\..*", "", rowname))
```

Comment:
Since you're using the same `gsub` in both chunks, you could remove and keep after the if-else when returning the final counts.
Also, I think `sub` is suitable for this usecase since you're looking to substitute first occurrences.

Reply:
updated in https://github.com/umccr/RNAsum/commit/9dce6f9d4768450154fb0471b71588148ffb4db7